execute_until_build_succeeds.py

```
@@ -0,0 +1,137 @@
+from pydantic import ValidationError
+import time
+from refine_unit_test_code import refine_code_based_on_errors, parse_function_calls_until_success
+from execute_build_and_tests import execute_build_and_tests
+from update_project_file import update_project_file
+from agent_check_past_actions import check_actions
+from github_bot import *
```

Ruff (0.8.0)

`from github_bot import *` used; unable to detect undefined names

(F403)